client_test/virtualfund_benchmark_test.go

```diff
@@ -0,0 +1,93 @@
+package client_test
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/client/engine/chainservice"
+	"github.com/statechannels/go-nitro/client/engine/messageservice"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// TestBenchmark sets up three clients, then runs a virtual funding benchmark, printing the duration
+// to the screen.
+func TestBenchmark(t *testing.T) {
+
+	logFile := "virtualfund_benchmark_test.log"
+	truncateLog(logFile)
+
+	chain := chainservice.NewMockChain()
+	broker := messageservice.NewBroker()
+
+	clientAlice := setupClient(aliceKey, chain, broker, logFile)
+	clientBob := setupClient(bobKey, chain, broker, logFile)
+	clientIrene := setupClient(ireneKey, chain, broker, logFile)
+
+	directlyFundALedgerChannel(t, clientAlice, clientIrene)
+	directlyFundALedgerChannel(t, clientIrene, clientBob)
+
+	done := make(chan bool)
+
+	n := 3
+	for i := 0; i < n; i++ {
+		go benchmarkVirtualChannelCreation(t, clientAlice, clientBob, irene, done)
+	}
+
+	expect(t, done, n, time.Millisecond*1000)
+}
+
+func benchmarkVirtualChannelCreation(t *testing.T, alice, bob client.Client, irene types.Address, done chan bool) {
```

Comment:
We could add some godoc to this function.

Reply:
https://github.com/statechannels/go-nitro/pull/354/commits/3bdada1c2c5578fd3ce5ea081672f067b0311405